From 355c223e4117bb72d1de83ebcc236535376fbb32 Mon Sep 17 00:00:00 2001
From: Haim Boot <hayim@marvell.com>
Date: Mon, 12 Dec 2016 14:41:34 +0200
Subject: [PATCH 1171/2241] fix: rtc: a80x0: fix rtc drift

This patch fixes the RTC drift between System Clock ('date')
and RTC Clock ('hwclock') as the RTC clock was too slow.

The RTC requires external oscillator.
The problem was that, in 80x0, CP0 RTC clock was used while CP1 RTC was
the one which is connected (by package) to the oscillator.
CP0 RTC is not connected at all (though the IP can be accessed).

This patch enables CP0 RTC for 70x0 and CP1 RTC for 80x0.

(Note: armada-cp110-1.dtsi was not updated as the RTC node there
is already disabled).

Change-Id: Ieeb5d0256821ddbcb51c149fde8429388d30e053
Signed-off-by: Haim Boot <hayim@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/34581
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-7040.dtsi  | 3 +++
 arch/arm64/boot/dts/marvell/armada-8040.dtsi  | 3 +++
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi | 1 +
 3 files changed, 7 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040.dtsi b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
index f1df28a..07b0c0c 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040.dtsi
@@ -105,6 +105,9 @@
 					marvell,function = "gpio";
 				};
 			};
+			rtc@284000 {
+				status = "okay";
+			};
 			gpio1: gpio@440100 {
 			};
 			gpio2: gpio@440140 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8040.dtsi b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
index 8dea327..f139cc6 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040.dtsi
@@ -240,6 +240,9 @@
 					marvell,function = "spi1";
 				};
 			};
+			rtc@284000 {
+				status = "okay";
+			};
 		};
 		pcie@0x600000 {
 			msi-parent = <&gic_v2m0>;
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 3ce8c12..a53b49f 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -54,6 +54,7 @@ rtc@284000 {
 	compatible = "marvell,armada8k-rtc";
 	reg = <0x284000 0x20>, <0x284080 0x24>;
 	reg-names = "rtc", "rtc-soc";
+	status = "disabled";
 };
 
 pinctrl@440000 {
-- 
2.7.4

